packages/toolkit/config/webpack/modules.js

```diff
@@ -52,52 +53,83 @@ function shouldExclude(input, include) {
 	return /node_modules/.test(input);
 }
 
+const LINARIA_EXTENSION = '.linaria.module.css';
+const LINARIA_EXTENSION_REGEXP = /\.linaria\.module\.css/;
+
 module.exports = ({
 	isProduction,
 	isPackage,
 	defaultTargets,
 	projectConfig: { wordpress, hot, include },
 }) => {
 	const hasReactFastRefresh = hot && !isProduction;
+
+	// Provide a default configuration if there's not
+	// one explicitly available in the project.
+	const babelConfig = !hasBabelConfig()
+		? {
+				babelrc: false,
+				configFile: false,
+				sourceType: 'unambiguous',
+				plugins: [hasReactFastRefresh && require.resolve('react-refresh/babel')].filter(
+					Boolean,
+				),
+				presets: [
+					[
+						require.resolve('@10up/babel-preset-default'),
+						{
+							wordpress,
+							useBuiltIns: isPackage ? false : 'usage',
+							targets: defaultTargets,
+						},
+					],
+				],
+		  }
+		: {};
+
+	if (isPackageInstalled('@linaria/babel-preset') && !hasBabelConfig()) {
+		babelConfig.presets.push([
+			'@linaria',
+			{
+				babelOptions: {
+					babelrc: false,
+					configFile: false,
+					sourceType: 'unambiguous',
+					presets: [...babelConfig.presets],
+				},
+			},
+		]);
+	}
+
 	return {
 		rules: [
 			{
 				// Match all js/jsx/ts/tsx files except TS definition files
 				test: /^(?!.*\.d\.tsx?$).*\.[tj]sx?$/,
 				exclude: (input) => shouldExclude(input, include),
 				use: [
-					require.resolve('thread-loader'),
+					{
+						loader: require.resolve('./plugins/noop-loader'),
```

Comment:
thread-loader is incompatible with linaria (at least without any workarounds). I added a `noop-loader` just in case projects are manipulating `rules` and if simply remove we might introduce off-by-one errors.

I never really understood the benefits of thread-loader so I'm not concerned about us just ditching it.... I believe this is a relatively safe change that shouldn't be breaking but open to thoughts @Antonio-Laguna

Reply:
I think you're correct! I don't believe this is breaking! The only effect is that it _might_ (only might) reduce speed by a tiny bit given there are no extra workers.